tinydir.h

```diff
@@ -704,6 +704,17 @@ if (errno)
 	base_name =basename(file_name_buf);
 #endif
 
+	/* Special case: if the path is a root dir, open the parent dir as the file */
+	if (_tinydir_strlen(base_name) == 0)
```

Comment:
A few lines above you check for empty string like this: `drive_buf[0] == '\0'`. Maybe do the same here?